include/aws/common/allocator.h

```diff
@@ -110,6 +110,50 @@ int aws_mem_realloc(struct aws_allocator *allocator, void **ptr, size_t oldsize,
  * that we can leave unchanged on failure.
  */
 
+enum aws_mem_trace_level {
+    AWS_MEMTRACE_NONE = 0,   /* no tracing */
+    AWS_MEMTRACE_BYTES = 1,  /* just track allocation sizes and total allocated */
+    AWS_MEMTRACE_STACKS = 2, /* capture callstacks for each allocation */
+};
+
+#if defined(AWS_HAVE_EXECINFO) || defined(WIN32) || defined(__APPLE__)
+#    define AWS_MEMTRACE_STACKS_AVAILABLE
+#endif
+
+/*
+ * Wraps an allocator and tracks all external allocations. If aws_mem_trace_dump() is called
+ * and there are still allocations active, they will be reported to stderr.
+ * frames_per_stack is how many frames to store per callstack if AWS_MEMTRACE_STACKS is in use,
+ * otherwise it is ignored. 8 tends to be a pretty good number balancing storage space vs useful stacks.
+ * Returns the tracer allocator, which should be used for all allocations that should be tracked.
+ */
+AWS_COMMON_API
+struct aws_allocator *aws_mem_tracer_new(
```

Comment:
comment here says `stderr`, but actually it goes through logger at TRACE level.
Update comment, or actually printf to stderr

Reply:
Fixed.